src/inmanta/server/services/resourceservice.py

```diff
@@ -722,49 +731,6 @@ async def resource_deploy_done(
                     changes=changes_with_rvid,
                 )
 
-    async def propagate_resource_state_if_stale(
-        self,
-        connection: Connection,
-        env: data.Environment,
-        resource_ids: list[Id],
-        last_produced_events: datetime.datetime,
-        deploy_state: ResourceState,
-        propagate_last_produced_events: bool,
-        propagate_deploy_state: bool,
-    ) -> None:
-        if propagate_deploy_state or propagate_last_produced_events:
-            # lock out release version
-            await env.acquire_release_version_lock(connection=connection)
-            latest_version = await data.ConfigurationModel.get_version_nr_latest_version(env.id, connection=connection)
-
-            for resource_id in resource_ids:
-                if latest_version is not None and latest_version > resource_id.version:
-                    # we are stale, forward propagate our status
```

Comment:
Can you elaborate on why this can go now? The resource table still has a state field, which is still the authority on a resource's state until we move those checks to the read path, isn't it?

Reply:
I have to give this comment more thought. That'll be for Monday.